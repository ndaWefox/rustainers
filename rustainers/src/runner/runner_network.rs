use serde::{Deserialize, Serialize};

/// The network driver
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum Driver {
    /// Bridge
    Bridge,
    /// Host
    Host,
    /// Overlay
    Overlay,
    /// Ipvlan
    Ipvlan,
    /// Macvlan
    Macvlan,
    /// Null
    Null,
}

/// A runner network
///
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub struct RunnerNetwork {
    #[serde(alias = "name")]
    /// The network name
    pub name: String,
    #[serde(alias = "scope")]
    /// The network scope
    pub scope: Option<String>,
    #[serde(alias = "driver")]
    /// The network driver
    pub driver: Driver,
}

#[cfg(test)]
mod tests {

    use super::*;
    use assert2::let_assert;

    #[test]
    fn should_serde_network() {
        let json_stream = include_str!("../../tests/assets/docker_networks.jsonl");
        let stream = serde_json::Deserializer::from_str(json_stream).into_iter::<RunnerNetwork>();
        let networks = stream.collect::<Result<Vec<_>, _>>();
        let_assert!(Ok(data) = networks);
        insta::assert_debug_snapshot!(data);
    }

    #[test]
    fn should_serde_podman_network() {
        let json_stream = include_str!("../../tests/assets/podman_networks.jsonl");
        let stream = serde_json::Deserializer::from_str(json_stream).into_iter::<RunnerNetwork>();
        let networks = stream.collect::<Result<Vec<_>, _>>();
        let_assert!(Ok(data) = networks);
        insta::assert_debug_snapshot!(data);
    }
}
