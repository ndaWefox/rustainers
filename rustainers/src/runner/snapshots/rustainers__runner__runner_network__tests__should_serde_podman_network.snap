---
source: rustainers/src/runner/runner_network.rs
assertion_line: 59
expression: data
---
[
    RunnerNetwork {
        name: "my_network_01HTY22KXBHP2MVFPFSSK9CQCG",
        scope: None,
        driver: Bridge,
    },
    RunnerNetwork {
        name: "my_network_01HTY4NQF1GH56JNA8MXNN0ZMF",
        scope: None,
        driver: Bridge,
    },
    RunnerNetwork {
        name: "my_network_01HTY4NQSHF5Y6Q572X1G748ZJ",
        scope: None,
        driver: Bridge,
    },
    RunnerNetwork {
        name: "my_network_01HTY4T0FHGZF5ZYYXG9SJM69R",
        scope: None,
        driver: Bridge,
    },
    RunnerNetwork {
        name: "my_network_01HTY4T1SWV848BEX4VXAR9FVE",
        scope: None,
        driver: Bridge,
    },
    RunnerNetwork {
        name: "podman",
        scope: None,
        driver: Bridge,
    },
    RunnerNetwork {
        name: "tc_redpanda-single_01hty22k12cw846sk95pafkxpz_redpanda_network",
        scope: None,
        driver: Bridge,
    },
    RunnerNetwork {
        name: "tc_redpanda-single_01hty4npgg9c0x5d536qjc88tx_redpanda_network",
        scope: None,
        driver: Bridge,
    },
    RunnerNetwork {
        name: "tc_redpanda-single_01hty4sweerk2zmv3exqj4xzte_redpanda_network",
        scope: None,
        driver: Bridge,
    },
]
